cfgmgr/natmgr.cpp

```diff
@@ -0,0 +1,6391 @@
+/*
+ * Copyright 2019 Broadcom Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string.h>
+#include "logger.h"
+#include "producerstatetable.h"
+#include "macaddress.h"
+#include "natmgr.h"
+#include "exec.h"
+#include "tokenize.h"
+#include "converter.h"
+#include "shellcmd.h"
+#include "warm_restart.h"
+#include "ipaddress.h"
+#include "ipprefix.h"
+
+using namespace std;
+using namespace swss;
+
+/* NatMgr Constructor */
+NatMgr::NatMgr(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, const vector<string> &tableNames) :
+        Orch(cfgDb, tableNames),
+        m_cfgStaticNatTable(cfgDb, CFG_STATIC_NAT_TABLE_NAME),
+        m_cfgStaticNaptTable(cfgDb, CFG_STATIC_NAPT_TABLE_NAME),
+        m_cfgNatPoolTable(cfgDb, CFG_NAT_POOL_TABLE_NAME),
+        m_cfgNatBindingsTable(cfgDb, CFG_NAT_BINDINGS_TABLE_NAME),
+        m_cfgNatGlobalTable(cfgDb, CFG_NAT_GLOBAL_TABLE_NAME),
+        m_cfgInterfaceTable(cfgDb, CFG_INTF_TABLE_NAME),
```

Comment:
intfmgr is responsible for that table

Reply:
I have removed these lines, but NatMgr is also a consumer for the Interface Tables, to get the configured nat_zone value on any interface, which will be used as a mark value (zone +1 ) in iptables rules.